dali/kernels/imgproc/convolution/convolution_gpu.h

```diff
@@ -31,6 +31,15 @@ namespace kernels {
  * @brief Apply convolution with 1-channel `window` in specified axis.
  *
  * Operation is done by using GEMM and generating the matrix from convolution window on the fly
+ *
+ * CUTLASS allows to calculate GEMM: D = alpha * A * B + beta * C. To calculate convolution,
+ * we modify the GEMM to generate one of the input matrices on the fly based on the convolution
+ * window. Such matrix can be generated by applying the convolution window to an indentity matrix
+ * of a desired size.
+ * To the convolution we pass: alpha = `scale`, beta = 0 and D = C = `out`
```

Comment:
So C can be any and we can put there A as well, right?

Reply:
Probably we can. What we can't do, is have an actual in-place computation with one of the inputs to multiplication.